### ISSUE TITLE
test_write_heka_header fails on 32-bit architectures

### ISSUE THREAD

**Issue by sathieu** [mozilla-services/lua_sandbox#168]
Since test_write_heka_header was introduced (b8f2516a83958a921d5f59b040cee2a0e3253c03 in 1.2.0), the Debian packages FTBFS (fail to build from source) on all 32-bit archs:
```
 7/14 Test  #7: test_heka_message ..................***Failed    0.00 sec
line: 345 (hlen == 5 + i) i: 4 received 5
Tests run: 9
```

Complete build logs at: https://buildd.debian.org/status/package.php?p=lua-sandbox

**Comment by sathieu** [comment]
Also, mips and s390x has those additionnal failures:
```
  7/14 Test  #7: test_heka_message ..................***Failed    0.00 sec
line: 247 (v.u.d == 1) invalid value: 3.03865e-319
Tests run: 7

      Start  8: test_heka_message_matcher
 8/14 Test  #8: test_heka_message_matcher ..........***Failed    0.01 sec
line: 133 (lsb_eval_message_matcher(mm, &m)) Fields[double] == 99.9
Tests run: 3

[...]
      Start 11: test_heka_sandbox
11/14 Test #11: test_heka_sandbox ..................***Failed    0.05 sec
tests and results are mis-matched
test: 3
expected: \x0a\x10\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10\x03\x22\x03\x69\x69\x6d\x4a\x02\x68\x6e\x52\x13\x0a\x06\x6e\x75\x6d\x62\x65\x72\x10\x03\x39\x00\x00\x00\x00\x00\x00\xf0\x3f\x52\x2c\x0a\x07\x6e\x75\x6d\x62\x65\x72\x73\x10\x03\x1a\x05\x63\x6f\x75\x6e\x74\x3a\x18\x00\x00\x00\x00\x00\x00\xf0\x3f\x00\x00\x00\x00\x00\x00\x00\x40\x00\x00\x00\x00\x00\x00\x08\x40\x52\x0e\x0a\x05\x62\x6f\x6f\x6c\x73\x10\x04\x42\x03\x01\x00\x00\x52\x0a\x0a\x04\x62\x6f\x6f\x6c\x10\x04\x40\x01\x52\x10\x0a\x06\x73\x74\x72\x69\x6e\x67\x22\x06\x73\x74\x72\x69\x6e\x67\x52\x15\x0a\x07\x73\x74\x72\x69\x6e\x67\x73\x22\x02\x73\x31\x22\x02\x73\x32\x22\x02\x73\x33
received: \x0a\x10\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x10\x03\x22\x03\x69\x69\x6d\x4a\x02\x68\x6e\x52\x13\x0a\x06\x6e\x75\x6d\x62\x65\x72\x10\x03\x39\x3f\xf0\x00\x00\x00\x00\x00\x00\x52\x2c\x0a\x07\x6e\x75\x6d\x62\x65\x72\x73\x10\x03\x1a\x05\x63\x6f\x75\x6e\x74\x3a\x18\x3f\xf0\x00\x00\x00\x00\x00\x00\x40\x00\x00\x00\x00\x00\x00\x00\x40\x08\x00\x00\x00\x00\x00\x00\x52\x0e\x0a\x05\x62\x6f\x6f\x6c\x73\x10\x04\x42\x03\x01\x00\x00\x52\x0a\x0a\x04\x62\x6f\x6f\x6c\x10\x04\x40\x01\x52\x10\x0a\x06\x73\x74\x72\x69\x6e\x67\x22\x06\x73\x74\x72\x69\x6e\x67\x52\x15\x0a\x07\x73\x74\x72\x69\x6e\x67\x73\x22\x02\x73\x31\x22\x02\x73\x32\x22\x02\x73\x33
1479458619 [3] iim lua/iim.lua:38: test: 4 err: inject_message() failed: rejected by the callback
line: 594 (8 == stats.im_cnt) received 0
Tests run: 13
```

And

**Comment by sathieu** [comment]
It looks like lsb_const_string.len should be "unsigned long long" like a lot of size_t occurences.

**Comment by trink** [comment]
@sathieu size_t is correct (as it matches the type returned by the string functions).  I will try to make a alternate platform cleanup pass in the next sprint.  In the mips case there are some todo's already in the code and will most likely remain open.

**Comment by sathieu** [comment]
What about doing this in the meantime:
``` patch
diff --git a/src/util/test/test_heka_message.c b/src/util/test/test_heka_message.c
index 2a5aa3e..3b050a4 100644
--- a/src/util/test/test_heka_message.c
+++ b/src/util/test/test_heka_message.c
@@ -340,7 +340,7 @@ static char* test_write_heka_header()
   char header[LSB_MIN_HDR_SIZE];
   size_t hlen;
 
-  for (unsigned i = 0; i < 7; ++i) {
+  for (unsigned i = 0; i < sizeof(void*)-1; ++i) {
     hlen = lsb_write_heka_header(header, 1LL << (i * 8));
     mu_assert(hlen == 5 + i, "i: %u received %" PRIuSIZE, i, hlen);
   }
```

Heka message length is limited to 32bit anyway (https://hekad.readthedocs.io/en/v0.10.0/message/index.html#stream-framing).

**Comment by trink** [comment]
Closed in #172
